packages/core/src/client.ts

```diff
@@ -1885,7 +1886,17 @@ export class MedplumClient extends EventTarget {
       xhr.setRequestHeader('Cache-Control', 'no-cache, no-store, max-age=0');
       xhr.setRequestHeader('Content-Type', contentType);
       xhr.setRequestHeader('X-Medplum', 'extended');
-      xhr.send(data);
+      if (data instanceof Readable) {
```

Comment:
Instead of `Readable` and `Writable` which are Node APIs, can we use `ReadableStream` and `WritableStream` since these will work on both browser and in Node?

https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream
https://developer.mozilla.org/en-US/docs/Web/API/WritableStream

They function a bit differently internally but you can easily convert to and from `node.Readable` and `ReadableStream` via some utils in `node:stream`, namely `Readable.fromWeb(ReadableStream)` and `Readable.toWeb(node.Readable)`.

https://nodejs.org/api/stream.html#streamreadabletowebstreamreadable-options

You can call `Readable.toWeb()` on your `Readable` before passing in the `ReadableStream` in as the `BinarySource`

I think it would be great to have file streaming on both the browser and Node platforms

Reply:
Ah yes that makes more sense
Moved that direction in this commit https://github.com/medplum/medplum/pull/3411/commits/b27e68c30fd41c42ba0fb73ea2886ab484404732